lib/daru/vector.rb

```diff
@@ -879,41 +879,56 @@ def to_s
       to_html
     end
 
-    # Create a summary of the Vector using Report Builder.
-    def summary(method=:to_text)
-      ReportBuilder.new(no_title: true).add(self).send(method)
-    end
-
-    # :nocov:
-    def report_building b # rubocop:disable Metrics/AbcSize,Metrics/MethodLength
-      b.section(name: name) do |s|
-        s.text "n :#{size}"
-        s.text "n valid:#{count_values(*Daru::MISSING_VALUES)}"
-        if @type == :object
-          s.text  "factors: #{factors.to_a.join(',')}"
-          s.text  "mode: #{mode}"
-
-          s.table(name: 'Distribution') do |t|
-            frequencies.sort_by(&:to_s).each do |k,v|
-              key = @index.include?(k) ? @index[k] : k
-              t.row [key, v, ('%0.2f%%' % (v.quo(count_values(*Daru::MISSING_VALUES))*100))]
-            end
-          end
+    # Create a summary of the Vector
+    def summary(indent_level=0) # rubocop:disable Metrics/AbcSize,Metrics/MethodLength
+      summary = '  =' * indent_level + "= #{name}"
+      summary << "\n  n :#{size}"
+      summary << "\n  n valid:#{count_values(*Daru::MISSING_VALUES)}"
+      if @type == :object
+        summary << "\n  factors: #{factors.to_a.join(',')}"
+        summary << "\n  mode: #{mode}"
+
+        # Display in table format
+        summary << '  ' * indent_level + '  Distribution'
+        key = []
+        val = []
+        nval = []
+        frequencies.sort_by(&:to_s).each do |k,v|
```

Comment:
I think let's do write some specs -- and then we'll immediately will see/document, what it outputs and in which order, and will be able to discuss which approach is reasonable.